both/api/sources/server/stream-chain/stream-types/transform-data/index.js

```diff
@@ -0,0 +1,98 @@
+import { check } from 'meteor/check';
+import { SimpleSchema } from 'meteor/aldeed:simple-schema';
+import { _ } from 'lodash';
+import entries from '/both/lib/entries';
+import getContext from './get-context';
+import vm from 'vm';
+
+const { Transform } = Npm.require('zstreams');
+
+function compileMapping(fieldName, javascript, context) {
+  const code = `(d) => (${javascript})`;
+
+  return vm.runInContext(code, context);
```

Comment:
> can't the function access its prototype at runtime, allowing it to change stuff outside the VM?

This would be a bit surprising - but I'll make sure to test it.